clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java

```diff
@@ -82,9 +82,9 @@
  * are retained only until either a new sequence number is written by the corresponding producer or the producerId
  * is expired from lack of activity.
  *
- * There is no similar need to preserve the timestamp from the original batch after compaction. The FirstTimestamp
+ * There is no similar need to preserve the timestamp from the original batch after compaction. The BaseTimestamp
  * field therefore always reflects the timestamp of the first record in the batch. If the batch is empty, the
- * FirstTimestamp will be set to -1 (NO_TIMESTAMP).
+ * BaseTimestamp will be set to -1 (NO_TIMESTAMP).
```

Comment:
The comment is still inaccurate, no? The base timestamp does not always reflect the timestamp of the first record in the batch.

Reply:
you are right, I've revised the comments to be accurate about BaseTimestamp after compaction.